ls1intum/Artemis
# `Development`: Update server dependencies

### Checklist
#### General
#### Server

### Motivation and Context

Updating Spring Boot to the latest version (2.7.0) and other dependencies. 

### Description

The update of Spring Boot broke a few integration tests because of the queries that were being used. However, the issue seems to have been whitespace or formatting related so the queries in Text/Quiz/ProgrammingExerciseRepository were just fixed with the formatting used in ModelingExerciseRepository which didn't have the same issue. I'm still unsure why the queries were actually broken, can someone maybe comment on that?

### Steps for Testing

Prerequisites:

Course A with at least one each of Text, Modeling, Quiz and Programming exercise.
Exam/Exercise group for course A with at least one each of Text, Modeling, Quiz and Programming exercise.
Course B with at least one each of Text, Modeling, Quiz and Programming exercise.
Exam/Exercise group for course B with at least one each of Text, Modeling, Quiz and Programming exercise.

Instructor 1 for course A and course B
Instructor 2 for course A **(and not course B)**

Steps:

With instructor 1:

1. Log in as instructor 1
2. Go to exercises page of any course you have access to
3. Click the "import quiz exercise" button. Verify that you can see (at least) all quiz exercises for course A and course B.
4. Use the search field and verify that it shows only exercises/course with the name in the search field.
5. Repeat steps 3 and 4 for the other exercise types (modeling/text/programming

With instructor 2:

1. Log in as instructor 2
2. Go to exercises page of any course you have access to
3. Click the "import quiz exercise" button. Verify that you can see (at least) all quiz exercises for course, **but not exercises for course B**.
4. Use the search field and verify that it shows only exercises/course with the name in the search field.
5. Repeat steps 3 and 4 for the other exercise types (modeling/text/programming


### Review Progress

#### Code Review
#### Manual Tests

### Screenshots
